### ISSUE TITLE
API proposal: special-cased whitelisting for certain attributes?

### ISSUE THREAD

**Issue by bradvogel** [guardian/html-janitor#29]
Hi, would you be open to allowing specific whitelisting rules for certain attributes? It's common to want to want to allow certain styles or css classes.

So the rule:

```
  font: {
    style: {
       background-color: true,
       color: true
    },
   'class': {
      myclass: true
   }
  }
```

Would turn: `<font class="myclass otherclass" style="color: #FFF; font-weight: 600">` into `<font class="myclass" style="color: #FFF">`.

So HTMLJanitor would basically have special-cased rules for certain attributes, such as 'style' and 'css' that happen to have multi-part values. There might be some others.

Let me know if you're open to this and I can file a PR.

**Comment by bradvogel** [comment]
ping @rrees

**Comment by rrees** [comment]
Yeah, this seems reasonable. I'm not sure we have this detailed a use-case so I didn't feel qualified to comment. Feel free to build the PR.

From a code point of view I think this set of functionality might make the single spec file unmanageable so it might be sensible to have all the rule-based functionality out in its own spec.
